interfaces/builtin/personal_files.go

```diff
@@ -0,0 +1,72 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+	"strings"
+)
+
+const personalFilesSummary = `allows access to personal files or directories`
+
+const personalFilesBaseDeclarationSlots = `
+  personal-files:
+    allow-installation:
```

Comment:
the plan is still to have allow-installation: false here, cc @jdstrand

Reply:
What we want is for personalFilesBaseDeclarationPlugs to have 'allow-installation: false' but personalFilesBaseDeclarationSlots is scoped to slot-snap-type: core. As such, this LGTM.